<a href="https://colab.research.google.com/github/mucioosorio/R-desde-cero/blob/main/Copia_de_R_desde_cero_Variaci%C3%B3nMayo_31.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 2. Dispersión, variabilidad o heterogeneidad

Estudiar la distribución de los valores de la serie, analizando si estos se encuentran más o menos concentrados, o más o menos dispersos. 

![tinystats.github.io](https://tinystats.github.io/teacups-giraffes-and-statistics/images/04_variance/giraffe_variance2.jpg)

***
> ¿Qué tan homogeneas son las exhalaciones del geyser Old Faithful?
***
Existen diversas medidas de dispersión, entre las más utilizadas podemos destacar las siguientes:

## Amplitud.
Es la medida de dispersión más fácil de obtener. Se obtiene hallando la diferencia entre la observación mayor y la menor (el valor máximo menos el valor mínimo). La amplitud es una medida muy influenciada por los valores extremos y por lo tanto puede dar una impresión falsa sobre los valores reales de la distribución.




In [ ]:
# La amplitud del tiempo que tardan las exhalaciones y el  tiempo entre exhalaciones
max(faithful$eruptions)-min(faithful$eruptions)
# la función range proporciona el mínimo y el máximo
range(faithful$eruptions)
# diferencia
diff(range(faithful$eruptions))

[1] 3.5

[1] 1.6 5.1

[1] 3.5

In [ ]:
c <- c(1,3,6,9,2)
diff(c)

[1]  2  3  3 -7

Interpretación:

## Varianza.
Mide la distancia (cuadrada) existente entre los valores de la serie y la media. Mientras más se aproxima a cero, más concentrados están los valores de la serie alrededor de la media. Por el contrario, mientras mayor sea la varianza, más dispersos están. El símbolo para representar a la varianza muestral es
$s^2$ y para la varianza poblacional $\sigma^2$.

$$s^2=\frac{1}{n-1}{\sum_{i=1}^{n}\left(x_i-\bar{x}\right)^2}$$

$$\sigma^2=\frac{1}{N}{\sum_{i=1}^{N}\left(x_i-\mu\right)^2}$$

## Desviación estándar o típica.
Se calcula como raíz cuadrada de la varianza.

Nótese que ni la varianza ni la desviación estándar pueden ser negativas y son cero sólo cuando todos los datos tienen el mismo valor que no es otro que el valor de la media.


In [ ]:
# La varianza de las exhalaciones y el tiempo entre exhalaciones
var(faithful$eruptions)
mean(faithful$eruptions)

[1] 1.302728

[1] 3.487783

Interpretación: Las exhalaciones se alejan de 3.48 min, aprox 1.3 min^2   NO SENTIDO

In [ ]:
# desviación estándar
sd(faithful$eruptions)

[1] 1.141371

Interpretación: Las exhalaciones se alejan de 3.48 min, aprox 1.14 min



* Se puede usar a la amplitud para estimar a la desviación estándar ([Range Rule for Standard Deviation](https://scholar.rose-hulman.edu/cgi/viewcontent.cgi?article=1211&context=rhumj)).

* [Estimación de la media y de la desviación estándar a partir de la amplitud](https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-14-135)

* [Use of the Range Instead of the Standard Deviation](https://www.jstor.org/stable/2281203).


In [ ]:
# Aproximaciones de la desviación estándar
# estimación de la desviación estándar
max(faithful$eruptions)- min(faithful$eruptions) / 4




[1] 4.7

In [1]:
# aproximación de la S a partir de la amplitu en los datos de 
# glucosa




# La [regla empirica](https://online.stat.psu.edu/stat200/lesson/2/2.2/2.2.7)
<center>

![](https://upload.wikimedia.org/wikipedia/commons/thumb/2/22/Empirical_rule_histogram.svg/553px-Empirical_rule_histogram.svg.png)

</center>

In [ ]:
# Regla empirica
# El 100% de las exhalaciones estarán entre
mean(faithful$eruptions)-3*sd(faithful$eruptions)
mean(faithful$eruptions)+3*sd(faithful$eruptions)


# pruebas 
shapiro.test(faithful$eruptions)


[1] 0.06366933

[1] 6.911897


	Shapiro-Wilk normality test

data:  faithful$eruptions
W = 0.84592, p-value = 9.036e-16


Interpretación:

In [ ]:
# En total hay 272 datos
# dim(faithful)

#empirico


   # si se cumple la regla empirica

# sin generar un dataframe



## Amplitud intercuartil

Es una medida resistente de variación,puesto que elimina al 25% de datos más pequeños y al 25% de los datos mas grandes mide la amplitud del 50% de los datos centrales. 

$$AIC=P_{0.75}-P_{0.25}$$

## Desviación absoluta de la mediana

Es la mediana de las desviaciónes absolutas de cada valor respecto a la mediana.



In [ ]:
# amplitud intercuartil
quantile(faithful$eruptions,0.75)-quantile(faithful$eruptions,.25)

75% 
2.2915

In [ ]:
# desviación absoluta de la mediana

# desv de cada dato de la mediana


In [ ]:
# comparación de las medidas de variación
# x = c(2, 4 ,8 ,9 ,11, 11, 12)



#y =c(2, 4, 8, 9, 11, 11, 120) 



## Compartir Notebooks a través de GitHub

[GitHub](github.com) es una plataforma de desarrollo colaborativo para alojar proyectos utilizando el sistema de control de versiones Git. Se utiliza principalmente para la creación de código fuente de programas de ordenador. 

En nuestro caso lo usaremos como un repositorio que nos permitirá compartir nuestras Notebooks con código de R y para crear una página web personal.
